src/cargo/util/rustc.rs

```diff
@@ -141,8 +141,8 @@ impl Rustc {
         self.cache.lock().unwrap().cached_success(cmd)
     }
 
-    pub fn push_wrapper(&mut self, wrapper: RustcWrapper) {
-        self.wrapper = Some(wrapper);
+    pub fn push_wrapper<T: Into<Option<RustcWrapper>>>(&mut self, wrapper: T) {
```

Comment:
You could in theory call `push_wrapper(None)` with this.... not sure if anybody ever will. If someone ever does though I want to be notified!

Reply:
I wonder if instead of `push_wrapper` and such, we could do something like this?

```rust
pub fn set_wrapper(&mut self, wrapper: ProcessBuilder) {
    assert!(self.wrapper.is_none());
    self.wrapper = Some(wrapper);
}
```

That way we don't have to duplicate args/env/etc, and I think there's only currently one wrapper so it's fine to avoid the term "push" because we're not adding more than one

Reply:
Yeah, additionally, multiple wrappers wouldn't work anyway, since we can't pass more than one wrapper down as the first argument (yet)

Reply:
The assert wont work thanks to the test fix::does_not_crash_with_rustc_wrapper which is expected to replace an already defined wrapper.